### PR TITLE
Init Account by tying it to near

### PR DESCRIPTION
**Pull request by Lev-Stambler**
- Add functionality which ties a user's Discord keys with their Near Account
- Add functionality to remove Discord keys from Near Account

In a future PR we gotta talk about a UX to go along with this, see #31

### CODE REVIEW

File: libs/near/src/web/wallet-connect.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const config = {
+    networkId: 'testnet',
+    keyStore,
+    nodeUrl: getRPCUrl(network),
+    walletUrl: getWalletUrl(network),
+    helperUrl: getHelperUrl(network),
+    explorerUrl: getExplorerUrl(network),
+  };
+  const near = await connect(config);
+  console.log(near);
```
Comment (Sladuca) [inline libs/near/src/web/wallet-connect.ts:24]: console log

File: libs/near/src/web/wallet-connect.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const wallet = new WalletConnection(near, 'Baf wallet?');
```
Comment (Sladuca) [inline libs/near/src/web/wallet-connect.ts:25]: why question mark?

**Review by Sladuca — CHANGES_REQUESTED** [review]
This is way simpler! Gonna be even simpler with one contract!

See comments

File: libs/near/src/web/wallet-connect.ts (hunk 2 above)
Comment (Lev-Stambler) [reply libs/near/src/web/wallet-connect.ts:25]: ??? lmao my b

File: apps/frontend/src/config/init.svelte [hunk 3]
```diff
@@ -4,17 +4,19 @@
...
 
   // TODO: we have to improve this to not require a reload, please see
   // https://github.com/bafnetwork/baf-wallet-v2/issues/29
   export async function reinitApp() {
     window.location.reload();
   }
-  export async function initApp() {
-    const { chainsState } = await initAccount();
-    if (checkChainInit(chainsState, Chain.NEAR))
+  export async function initApp(): Promise<{ initNear: boolean }> {
```
[the remark this answers is not in the document]
Comment (Lev-Stambler) [reply apps/frontend/src/config/init.svelte:14]: My bad, I will make sure to not do that in future

File: apps/frontend/src/state/accounts.svelte [hunk 4]
```diff
@@ -24,6 +29,7 @@
   export async function initAccount(): Promise<{
     accountState: AccountState;
     chainsState: ChainsState | null;
+    keys: KeyState | null;
```
Comment (Sladuca) [inline apps/frontend/src/state/accounts.svelte:32]: please move inline interface decl into a separate named interface

File: apps/frontend/src/state/chains.svelte [hunk 5]
```diff
@@ -21,16 +23,36 @@
...
+      secpPubkeyB58: secpPK.format(Encoding.BS58),
+    });
+    if (!associatedAccountId?.nearId) return false;
+    const chainAccount = await chainState[chain].accounts.lookup(
+      associatedAccountId.nearId
+    );
+    const associatedKeys = await chainState[chain].accounts.associatedKeys(
+      chainAccount
+    );
+    console.log(associatedKeys);
```
[the remark this answers is not in the document]
Comment (Sladuca) [reply apps/frontend/src/state/chains.svelte:50]: add comment if you have a good reason for these to stay

File: libs/interfaces/src/lib/chains.ts [hunk 6]
```diff
@@ -112,6 +112,7 @@ export interface AccountsInterface<Account, LookupParams, CreateParams> {
   lookup: (params: LookupParams) => Promise<Account>;
   create: (params: CreateParams) => Promise<Account>;
   getGenericMasterAccount: () => GenericAccount;
+  associatedKeys: <T>(account: Account) => Promise<PublicKey<T>[]>;
```
[the remark this answers is not in the document]
Comment (Sladuca) [reply libs/interfaces/src/lib/chains.ts:115]: see this

**Review by Sladuca — CHANGES_REQUESTED** [review]
Address other changes, or at least comment why you disagree

**Comment by Lev-Stambler** [comment]
@Sladuca what do you mean by `see this`?

**Review by Sladuca — APPROVED** [review]
I'll fix the rest of the things, dw about it

**Pull request merged**